### PR TITLE
combine all_are_zero_lamports

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
combine shrink and ancient append vec code

#### Summary of Changes

Fixes #

### CODE REVIEW

File: runtime/src/accounts_db.rs [hunk 1]
```diff
@@ -3566,6 +3574,9 @@ impl AccountsDb {
                         result = AccountsIndexScanResult::Unref;
                         dead += 1;
                     } else {
+                        if all_are_zero_lamports && stored_account.account.lamports() != 0 {
```
Comment (apfitzge) [inline runtime/src/accounts_db.rs:3577]: We can do this without an if:
`all_are_zero_lamports &= (stored_account.account.lamports == 0)`

File: runtime/src/accounts_db.rs [hunk 2]
```diff
@@ -3686,6 +3699,9 @@ impl AccountsDb {
                         .unwrap()
                         .append(&mut unrefed_pubkeys);
                     alive_total_collect.fetch_add(alive_total, Ordering::Relaxed);
+                    if !all_are_zero_lamports {
```
Comment (apfitzge) [inline runtime/src/accounts_db.rs:3702]: can do similar here if we want.

**Review by apfitzge — APPROVED** [review]
Good change - could be a bit more concise using binary ops on the bools. I prefer that, but not going to block on it. Compiler should optimize any branching there anyway.

File: runtime/src/accounts_db.rs (hunk 1 above)
Comment (jeffwashington) [reply runtime/src/accounts_db.rs:3651]: done

File: runtime/src/accounts_db.rs (hunk 2 above)
Comment (jeffwashington) [reply runtime/src/accounts_db.rs:3776]: thjs involves a mutex though

**Pull request merged**